### DISCUSSION TITLE
Getting changed field value in the On Change event

### DISCUSSION THREAD

**Discussion by xlog-null** [Budibase/budibase#7638] [Help] [answered]
Hi, not sure if this is a bug but I have an options picker and in it's On Change event, I am using JavaScript to try to get the value that the field has changed to. But it looks like inside the event, when I get the value of the field, it returns the field that was set before On Change was fired. Surely, when the On Change event is fired, getting the field value from inside my function, the new value should be returned?

**Comment by avivkri** [comment, marked as answer]
Hi @xlog-null,

To access the new or the changed value of the field, you will have to use either `{{ Field Value }}` in Handlebars or `$("Field Value")` in Javascript. This is described in the [Actions documentation page](https://docs.budibase.com/docs/actions#update-field-value), although it was difficult to find it the first time, mistaking it for a bug.

> In addition, there is a convenient Binding {{ Field Value }} which represents the value of the trigger form field.

So, accessing the field directly like `{{ someForm.Fields.someField }}` will return the value before onChange event
and accessing the field like `{{ Field Value }}` will return the new or the changed value.

I hope you find this helpful.

**Comment by xlog-null** [comment]
Hi, thank you for the reply, however, accessing the field value with `Value` or without produces the exact same result.

This is the JavaScript in my On Change function:
```
console.log($("Form.Fields.severity"));
console.log($("Form.Fields.severity Value"));
```

I have these options in the options picker box:
![image](https://user-images.githubusercontent.com/92792093/188579932-cf3ee0ed-1ea8-4375-a5c3-296e709ee572.png)

Changing the value from x to y will always yield the value is x. The output of the `console.log()` calls when changing from `High` to `Medium` is two lines with the value: `High`. Then, if I change from `Medium` to anything else the output will be to lines saying `Medium`.

Thanks

**Comment by xlog-null** [comment]
Oh my.... I have realised that the binding `Field Value`'s syntax is exactly that - `Field Value`. I thought `Field` was just a shorthand for `Form.Fields.field`... wow!

Thanks very much!
